src/codex_plus/subagents/__init__.py

```
@@ -0,0 +1,723 @@
+                return False
+        
+        # Check for network operations
+        if re.search(r'\b(fetch|download|api|http|request)', task.lower()):
+            if AgentCapability.NETWORK_ACCESS not in self.capabilities:
+                return False
+        
+        return True
+    
+    async def _execute_task(self, task: str, context: AgentContext) -> str:
+        """Execute the actual task. Override in subclasses."""
+        cli_callable = getattr(self, "_execute_via_claude_cli")
```

Ruff (0.13.1)

Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)